### PR TITLE
Add filtering and ordering on cars

### PR DESCRIPTION
**Pull request by Bogdan-Strat**

### CODE REVIEW

File: src/main/java/ro/unibuc/hello/controller/CarController.java [hunk 1]
```diff
@@ -30,13 +30,12 @@ public CarsDTO getCar(@RequestParam(name="carId") String id) {
 
     @PostMapping("/car/insert")
     @ResponseBody
-    public CarsDTO insertCar(@RequestParam(name="carId") String carId,
-                            @RequestParam(name="carMaker") String carMaker,
+    public CarsDTO insertCar(@RequestParam(name="carMaker") String carMaker,
```
Comment (sebastianiene) [inline src/main/java/ro/unibuc/hello/controller/CarController.java:33]: I see you dropped the "carId", I guess we are fine, as long as we have that @Id that increments automatically.

**Review by sebastianiene — APPROVED** [review]
approved the pull request!

File: src/main/java/ro/unibuc/hello/service/CarService.java [hunk 2]
```diff
@@ -78,4 +78,29 @@ public String deleteCar(String id){
...
+        List<CarsDTO> carsMapped = new ArrayList<> ();
+        carRepository.findAll().forEach(x -> carsMapped.add(new CarsDTO(x)));
+
+         carsMapped.sort(Comparator.comparing(CarsDTO::getCarPrice));
+
+         return carsMapped;
+
+    }
+
+    public List<CarsDTO>  filterCarsByCarTypeFilterCarsByCarType(String carType){
```
Comment (mirunaavram) [inline src/main/java/ro/unibuc/hello/service/CarService.java]: denumirea nu este intuitiva

File: src/main/java/ro/unibuc/hello/service/CarService.java [hunk 3]
```diff
@@ -78,4 +78,29 @@ public String deleteCar(String id){
...
+        List<CarEntity> cars = carRepository.findAll();
+        List<CarsDTO> carsMapped = new ArrayList<> ();
+
+        for(int i = 0; i < cars.size(); i++){
+            if (Objects.equals(cars.get(i).getCarType(), carType)) {
```
Comment (mirunaavram) [inline src/main/java/ro/unibuc/hello/service/CarService.java]: eu as fi folosit foreach

File: src/main/java/ro/unibuc/hello/controller/CarController.java (hunk 1 above)
Comment (Bogdan-Strat) [reply src/main/java/ro/unibuc/hello/controller/CarController.java:33]: the Id annotation in CarEntity generate a guid for every entity insert

File: src/main/java/ro/unibuc/hello/service/CarService.java (hunk 2 above)
Comment (Bogdan-Strat) [reply src/main/java/ro/unibuc/hello/service/CarService.java]: done

File: src/main/java/ro/unibuc/hello/service/CarService.java (hunk 3 above)
Comment (Bogdan-Strat) [reply src/main/java/ro/unibuc/hello/service/CarService.java]: done

**Review by mirunaavram — APPROVED** [review]

**Pull request merged**